relbench/tasks/trial.py

```diff
@@ -0,0 +1,296 @@
+                LEFT JOIN studies s
+                ON s.nct_id = o.nct_id 
+                where (oa.p_value_modifier is null or oa.p_value_modifier != '>')
+                and oa.p_value >=0
+                and oa.p_value <=1
+                and o.outcome_type = 'Primary'
+            )
+
+            SELECT
+                t.timestamp,
+                tr.nct_id,
+                CASE 
+                    WHEN MIN(tr.p_value) <= 0.05 THEN 1
+                    ELSE 0
+                END AS outcome
+            FROM timestamp_df t
+            LEFT JOIN TRIAL_INFO tr
+            ON tr.start_date <= t.timestamp
+                and tr.date > t.timestamp
+                and tr.date <= t.timestamp + INTERVAL '{self.timedelta}'
+            GROUP BY t.timestamp, tr.nct_id;
+            """ 
+        ).df()
+
+        return Table(
+            df=df,
+            fkey_col_to_pkey_table={self.entity_col: self.entity_table},
+            pkey_col=None,
+            time_col=self.time_col,
+        )
+
+
+class AdverseEventTask(RelBenchTask):
+    r"""Predict the number of affected patients with severe advsere events/death for the trial."""
+
+    name = "rel-trial-adverse"
+    task_type = TaskType.REGRESSION
+    entity_col = "nct_id"
+    entity_table = "studies"
+    time_col = "timestamp"
+    target_col = "num_of_adverse_events"
+    timedelta = pd.Timedelta(days=365)
+    metrics = [mae, rmse]
+
+    def make_table(self, db: Database, timestamps: "pd.Series[pd.Timestamp]") -> Table:
+        timestamp_df = pd.DataFrame({"timestamp": timestamps})
+        reported_event_totals = db.table_dict["reported_event_totals"].df
+        studies = db.table_dict["studies"].df
+        
+        df = duckdb.sql(
+            f"""
+            WITH TRIAL_INFO AS (
+                SELECT
+                    r.nct_id,
+                    r.event_type,
+                    r.subjects_affected,
+                    r.date,
+                    s.start_date
+                FROM reported_event_totals r
+                LEFT JOIN studies s
+                ON r.nct_id = s.nct_id
+                WHERE r.event_type = 'serious' or r.event_type = 'deaths'
+            )
+        
+            SELECT
+                t.timestamp,
+                tr.nct_id,
+                sum(tr.subjects_affected) AS num_of_adverse_events
+            FROM timestamp_df t
+            LEFT JOIN TRIAL_INFO tr
+            ON tr.start_date <= t.timestamp
+                and tr.date > t.timestamp
+                and tr.date <= t.timestamp + INTERVAL '{self.timedelta}'
+            GROUP BY t.timestamp, tr.nct_id;
+            """ 
+        ).df()
+
+        return Table(
+            df=df,
+            fkey_col_to_pkey_table={self.entity_col: self.entity_table},
+            pkey_col=None,
+            time_col=self.time_col,
+        )
+
+
+class WithdrawalTask(RelBenchTask):
+    r"""Predict the the set of reasons of withdrawals for each trial"""
```

Comment:
Can we make this task description temporal? predict ... for the next XXX days/year

Reply:
Same for the other task descriptions.